server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -463,12 +463,18 @@ public void consumerExists(
     @Path("exists")
     public Response consumerExistsBulk(Iterable<String> consumerUuids) throws BadRequestException {
         if (consumerUuids != null && consumerUuids.iterator().hasNext()) {
-            Set<String> nonExistingUuids = consumerCurator.filterNonExistingConsumerIds(consumerUuids);
+            Set<String> existingUuids = consumerCurator.getExistingConsumerUuids(consumerUuids);
+
+            Set<String> nonExistingUuids = new HashSet<>();
+            consumerUuids.forEach(nonExistingUuids::add);
+            nonExistingUuids.removeAll(existingUuids);
```

Comment:
Do we need the extra set here? You can just remove the received set from the input set and get the same result without doing a copy.

Reply:
Here I don't have input set rather its `Iterable` type (it does not have remove method).  Even If I change it to `Collection` type I don't have a way to remove duplicate and return only unique IDs

Reply:
Collection does have remove methods, but they're optional, so it's a bit dangerous to rely on them.

In this case since it's not a global or utility method, nor something that is commonly used elsewhere, it's not critical to be generic with your inputs\*. Also, we know RESTEasy is nice enough to give us whatever collection type we request.

\* My comment regarding that was in the context of libraries and shared classes. Sorry for the confusion.

Reply:
Changed the Inputs to accept `SET`